frontend/src/components/Auth/AuthContext.js

```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from 'react';
+import app from '../Firebase';
+
+const AuthContext = React.createContext();
+
+/**
+ * AuthProvider component that keeps track of the authentication status of the
+ * current user. Allows global use of this status using React Context and should
+ * be wrapped around the contents of the App component.
+ */
+const AuthProvider = (props) => {
+  const [currentUser, setCurrentUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
I did try writing it as a class component originally, since Hooks do have an equivalent that can be written in classes, but I was getting weird bugs that I couldn't figure out. Using React Hooks made it both easier to use and cleaner to read, so I highly recommend using them if you find it can make your code better.